scipy/sparse/linalg/isolve/iterative.py

```diff
@@ -492,6 +492,8 @@ def gmres(A, b, x0=None, tol=1e-5, restart=None, maxiter=None, M=None, callback=
           - ``legacy`` (default): same as ``pr_norm``, but also changes the
             meaning of 'maxiter' to count inner iterations instead of restart
             cycles.
+    orth : 'mgs', optional
```

Comment:
Actually, what does `orth` means? `method` would be clearer. Also the type should be `str`. Then you would put the options in the description of the parameter as well as the default value.

Reply:
Okay. `orth` represents what orthogonalization method to use in Arnoldi process (in line 946-951). Different orthogonalization method can be used here to enhance the robustness and effciency. I use Modified Gram-Schmidt orthogonalization.

Reply:
Ok, then this description should actually be in the description of the parameter. Do you plan to add more options, than the current (which needs to have a name) one and `mgs`?

Reply:
Yes, After that, I will add Householder orthogonalization and named as `orth=householder`.